fv3core/tests/main/test_dycore_call.py

```diff
@@ -0,0 +1,161 @@
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_v,
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_w,
+        dynamical_core._lagrangian_to_eulerian_obj._map_single_delz,
+    ):
+        original_attributes[obj] = obj._lagrangian_contributions
+        obj._lagrangian_contributions = do_nothing  # type: ignore
+    for (
+        obj
+    ) in dynamical_core._lagrangian_to_eulerian_obj._mapn_tracer._list_of_remap_objects:
+        original_attributes[obj] = obj._lagrangian_contributions
+        obj._lagrangian_contributions = do_nothing  # type: ignore
+    try:
+        yield
+    finally:
+        for obj, original in original_attributes.items():
+            obj._lagrangian_contributions = original
+
+
+def setup_dycore() -> Tuple[fv3core.DynamicalCore, List[Any]]:
+    backend = "numpy"
+    stencil_config = fv3core.StencilConfig(
+        backend=backend,
+        rebuild=False,
+        validate_args=True,
+    )
+    mpi_comm = NullComm(rank=0, total_ranks=6, fill_value=0.0)
+    namelist = Namelist.from_f90nml(f90nml.read(NAMELIST_FILENAME))
+    partitioner = pace.util.CubedSpherePartitioner(
+        pace.util.TilePartitioner(namelist.layout)
+    )
+    communicator = pace.util.CubedSphereCommunicator(mpi_comm, partitioner)
+    sizer = pace.util.SubtileGridSizer.from_tile_params(
+        nx_tile=namelist.npx - 1,
+        ny_tile=namelist.npy - 1,
+        nz=namelist.npz,
+        n_halo=3,
+        extra_dim_lengths={},
+        layout=namelist.layout,
+        tile_partitioner=partitioner.tile,
+        tile_rank=mpi_comm.rank,
+    )
+    grid_indexing = fv3core.GridIndexing.from_sizer_and_communicator(
+        sizer=sizer, cube=communicator
+    )
+    metric_terms = MetricTerms.from_tile_sizing(
+        npx=namelist.npx,
+        npy=namelist.npy,
+        npz=namelist.npz,
+        communicator=communicator,
+        backend=backend,
+    )
+
+    # create an initial state from the Jablonowski & Williamson Baroclinic
+    # test case perturbation. JRMS2006
+    state = baroclinic_init.init_baroclinic_state(
+        metric_terms,
+        adiabatic=namelist.adiabatic,
+        hydrostatic=namelist.hydrostatic,
+        moist_phys=namelist.moist_phys,
+        comm=communicator,
+    )
+    stencil_factory = fv3core.StencilFactory(
+        config=stencil_config,
+        grid_indexing=grid_indexing,
+    )
+
+    dycore = fv3core.DynamicalCore(
+        comm=communicator,
+        grid_data=GridData.new_from_metric_terms(metric_terms),
+        stencil_factory=stencil_factory,
+        damping_coefficients=DampingCoefficients.new_from_metric_terms(metric_terms),
+        config=namelist.dynamical_core,
+        phis=state.phis_quantity,
+    )
+    do_adiabatic_init = False
+    # TODO compute from namelist
+    bdt = 225.0
+
+    # TODO: lagrangian contributions currently cause an out of bounds iteration
+    # when halo updates are disabled. Fix that bug and remove this decorator.
```

Comment:
@jdahm , @elynnwu and I tried for a while to fix this bug but concluded this can only happen with an update to gt4py.